### PR TITLE
docs: move docs to `main` branch #969

### LINKED ISSUE #969: docs: bring the docs closer to sources
Currently, documentation generation is in a different `gh-pages` branch. Sometimes there is a problem when usage examples lag behind the current implementation of `main`

Let's try:
* Move the docs infra into `main`
* Let the docs publish task be triggered manually
* Update docs on every release

### PR DESCRIPTION
**Pull request by 1e9y**
Moving documentation from the `gh-pages` branch to `main` branch to keep it closer to the sources. What I have done in this PR:

- [x] Amended `docs` workflow to trigger on new release event.
- [x] Moved Vitepress configuration from branch (root) to main (`docs` folder)
- [x] Added relevant `package.json` commands
- [x] Moved all doc sources to main

Fixes #969

- [ ] Tests pass
- [x] Appropriate changes to README are included in PR

### CODE REVIEW

**Comment by 1e9y** [comment]
@antonmedv Can you please confirm that this solution aligns with your idea?

File: docs/.vitepress/config.mts [hunk 1]
```diff
@@ -0,0 +1,103 @@
+import { defineConfig } from 'vitepress'
+
+// https://vitepress.dev/reference/site-config
+export default defineConfig({
+  ignoreDeadLinks: true,
+  base: '/zx/',
+  outDir: 'docs',
```
Comment (antongolub) [inline docs/.vitepress/config.mts:7]: This does mean, that `<root>/docs/docs` will be created, while _Upload artifact_ step is still referencing to `<root>/docs`

**Review by antongolub — CHANGES_REQUESTED** [review]

File: docs/.vitepress/config.mts (hunk 1 above)
Comment (antonmedv) [reply docs/.vitepress/config.mts:7]: Lets rename it to build?

Comment (1e9y) [reply docs/.vitepress/config.mts]: Good point, @antongolub. I renamed the artifact directory to `build`, as @antonmedv suggested, and amended workflow file and Vitepress configuration accordingly.

**Comment by antongolub** [comment]
https://github.com/google/zx/actions/runs/12357969325/job/34491188714?pr=985
No idea what exactly is broken, but my first steps for similar cases: `rm -rf node_modules package-lock.json && npm i`

**Comment by 1e9y** [comment]
@antongolub 

My latest commit has a green tick. But it didn't run the workflow for some reason... I guess one of you must approve it.

**Review by antongolub — APPROVED** [review]
Seems good.

**Comment by antongolub** [comment]
@1e9y , 
Thanks for the contribution. Now you're able to update your gh profile badges to get the reward.

**Pull request merged**